compiler/luci/pass/src/CompressWeightsPass.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/CompressWeightsPass.h"
+#include "helpers/HuffmanEncoder.h"
+#include "helpers/NodeFiller.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Service/Nodes/CircleConst.h>
+
+#include <cmath>
+#include <cassert>
+
+namespace
+{
+template <loco::DataType T> class TypeSelector;
+
+template <> class TypeSelector<loco::DataType::U8>
+{
+public:
+  using Type = uint8_t;
+};
+template <> class TypeSelector<loco::DataType::S8>
+{
+public:
+  using Type = int8_t;
+};
+
+template <loco::DataType DT> bool compress_weights_huffman_conv2d(luci::CircleConv2D *conv2d)
+{
+  using T = typename TypeSelector<DT>::Type;
+  assert(conv2d);
+  if (conv2d->weightCompression() == luci::WeightCompression::HUFFMAN)
+    return false;
```

Comment:
maybe? 
```suggestion
  if (conv2d->weightCompression() != luci::WeightCompression::NONE)
    return false;
```
in case we had another algorithm and current conv is compressed in that algorithm

Reply:
or, we have to decompress that, and re-compress to huffman.